### PR TITLE
feat: add psbt functionality

### PR DESCRIPTION
**Pull request by edgarkhanzadian**
- Add PsbtSigner component.
- Hook up PsbtSigner to send flow
- Create a basic psbt with 1 input utxo for psbt testing
- Market data should be returned in cents, not dollars
- Fix flaky balances using unique react query keys for btc balance
- Update api to the leather-hosted api
- Fix encryption compliance in Info.plist so appstore doesn't require manual submission for every build
- Add testnet4 fee rate request

Note: I'll leave the commits unsquashed for now until right before the merge, just in case i need to revert some of the changes.

### CODE REVIEW

File: apps/mobile/src/features/approver/components/inputs-outputs-card.tsx
Comment (kyranjamie) [inline apps/mobile/src/features/approver/components/inputs-outputs-card.tsx:16]: https://www.notion.so/trustmachines/Naming-Component-Props-13e6ba7dd5ab4404a1cda0de6175cda6?pvs=4

File: apps/mobile/src/features/approver/components/utxo-row.tsx
Comment (kyranjamie) [inline apps/mobile/src/features/approver/components/utxo-row.tsx:22]: Also here, and `txid`

File: apps/mobile/src/queries/balance/bitcoin-balance.query.ts
Comment (kyranjamie) [inline apps/mobile/src/queries/balance/bitcoin-balance.query.ts:38]:
```suggestion
function useBitcoinDescriptorsByAcccountPaymentType(
```
?

File: apps/mobile/src/queries/balance/bitcoin-balance.query.ts
Comment (kyranjamie) [inline apps/mobile/src/queries/balance/bitcoin-balance.query.ts:47]: 👍🏼

**Review by kyranjamie — COMMENTED** [review]
Great work @edgarkhanzadian this is coming together 👌🏼

File: apps/mobile/src/features/approver/components/utxo-row.tsx
Comment (kyranjamie) [inline apps/mobile/src/features/approver/components/utxo-row.tsx:7]: three dots vs ellipsis 

... vs …

Comment (kyranjamie) [reply apps/mobile/src/features/approver/components/utxo-row.tsx:7]: also we have a truncateMiddle fn somewhere that does this

File: apps/mobile/src/features/approver/psbt/psbt-signer.tsx
Comment (kyranjamie) [inline apps/mobile/src/features/approver/psbt/psbt-signer.tsx:29]: should we change folder nesting to to just `feature/psbt-signer`?

File: packages/query/src/bitcoin/balance/btc-blockbook-utxos.ts
Comment (kyranjamie) [reply packages/query/src/bitcoin/balance/btc-blockbook-utxos.ts:31]: yep I saw a similar bug elsewhere

File: apps/mobile/src/components/spinner-icon.tsx
Comment (tigranpetrossian) [inline apps/mobile/src/components/spinner-icon.tsx:39]: nit: reverse -> invert

File: apps/mobile/src/features/approver/components/inputs-outputs-card.tsx
Comment (tigranpetrossian) [inline apps/mobile/src/features/approver/components/inputs-outputs-card.tsx:22]: I've seen this used in few other places, and it can potentially cause flicker from 0 to actual balance, unless the relevant query happens to be in cache and fresh. Is there anything we're missing design-wise to avoid this?
Couple things I can think of:
1. Loading state (if applicable)
2. Making shared data like market queries globally available upon app startup

File: apps/mobile/src/features/send/send-form/hooks/use-send-form-btc.tsx
Comment (tigranpetrossian) [inline apps/mobile/src/features/send/send-form/hooks/use-send-form-btc.tsx:19]:
```suggestion
const SATS_IN_BTC = 100_000_000
```
Can we put this in the `constants` package?

**Review by tigranpetrossian — APPROVED** [review]

File: apps/mobile/src/features/approver/components/fee-card.tsx
Comment (fbwoolf) [inline apps/mobile/src/features/approver/components/fee-card.tsx:24]: Maybe use a props interface?

File: apps/mobile/src/features/send/send-form/hooks/use-send-form-btc.tsx
Comment (fbwoolf) [inline apps/mobile/src/features/send/send-form/hooks/use-send-form-btc.tsx:46]: Was this all just for testing? Please don't merge over my work in latest btc send form PR to gen the unsigned psbt.

**Comment by fbwoolf** [comment]
Great work @edgarkhanzadian, can you rebase on my latest merge before I approve? My changes in `onInitSendTransfer` in `use-send-form.tsx` should generate the unsigned psbt to pass to your work.

File: packages/services/src/market-data/market-data.service.ts
Comment (pete-watters) [reply packages/services/src/market-data/market-data.service.ts:84]: Thanks @edgarkhanzadian . @alexp3y was going to fix this in https://linear.app/leather-io/issue/LEA-1761/stx-market-data-returns-incorrect-precision

File: apps/mobile/ios/leatherwalletmobile/Info.plist
Comment (pete-watters) [reply apps/mobile/ios/leatherwalletmobile/Info.plist:6]: Thanks @edgarkhanzadian 👍  Maybe we can take this change into a smaller PR so we can get it merged sooner? 

We didn't do another Testflight release since last week and there are some bug fixes we need to get out pre-Leatherhood.

File: apps/mobile/src/components/spinner-icon.tsx
Comment (pete-watters) [inline apps/mobile/src/components/spinner-icon.tsx:17]: Maybe we can move this to the UI package and keep the assets there too?

File: apps/mobile/src/features/approver/components/approver-account-card.tsx
Comment (pete-watters) [inline apps/mobile/src/features/approver/components/approver-account-card.tsx:14]: How about moving `AccountItem` into it's own file?

File: apps/mobile/src/features/approver/components/inputs-outputs-card.tsx
Comment (pete-watters) [reply apps/mobile/src/features/approver/components/inputs-outputs-card.tsx:22]: We have been hardcoding this fallback to `0` as we had been awaiting design. That's finished now and updating this is part of https://linear.app/leather-io/issue/LEA-1726/handle-balance-loading-state-elegantly 

We need to improve this so we never show `0` unless it is actually `0` and while it is loading to show a loader and if it fails then show a fallback. [The design is here ](https://www.figma.com/design/FlmF7ropcijOb7iVSDqNn0/%F0%9F%9F%AA-App-Explorations-2024-H2?node-id=22529-339198&node-type=frame&t=QXcdDievPkzQRpoE-0). 

Once we have a static UI for it we can backfill and refactor away the `createMoney(0, 'USD')`

File: apps/mobile/src/features/approver/components/outcomes-card.tsx
Comment (pete-watters) [inline apps/mobile/src/features/approver/components/outcomes-card.tsx:9]: Maybe this can be split into 3 component files? `BitcoinAddress`  + `BitcoinOutcome` + `OutcomeAddressesCard`?

File: apps/mobile/src/features/balances/bitcoin/bitcoin-balance.tsx
Comment (pete-watters) [inline apps/mobile/src/features/balances/bitcoin/bitcoin-balance.tsx:46]: Maybe we can set `px="5"` + `py="3"` in `TokenBalance` if it's almost always needed? 

Then it can be overwritten when it's not (like in `BitcoinOutcome` )

File: apps/mobile/src/features/psbt-signer/psbt-signer.tsx
Comment (pete-watters) [inline apps/mobile/src/features/psbt-signer/psbt-signer.tsx:30]: Lets call `net` `network` like we do elsewhere?

File: apps/mobile/src/features/psbt-signer/psbt-signer.tsx
Comment (pete-watters) [inline apps/mobile/src/features/psbt-signer/psbt-signer.tsx:75]: Should `requester` be hard-coded?

File: apps/mobile/src/features/psbt-signer/utils.ts
Comment (pete-watters) [inline apps/mobile/src/features/psbt-signer/utils.ts:24]: You can add a type for `psbtHex: string` and share it

**Comment by pete-watters** [comment]
Nice work on this Edgar 👏 It's really coming together. 

Maybe you could add some unit tests for your work in the BTC package? 

Also FYI @alexp3y is working on fixing issues with flaky balances and incorrect precision so you don't need to worry about that

File: apps/mobile/ios/leatherwalletmobile/Info.plist
Comment (pete-watters) [reply apps/mobile/ios/leatherwalletmobile/Info.plist]: I picked and merged this in https://github.com/leather-io/mono/pull/672

File: packages/services/src/market-data/market-data.service.ts
Comment (pete-watters) [reply packages/services/src/market-data/market-data.service.ts]: Alex fixed this in https://github.com/leather-io/mono/pull/671

**Pull request merged**
